bigbluebutton-client/src/org/bigbluebutton/modules/whiteboard/WhiteboardCanvasDisplayModel.as

```diff
@@ -442,66 +442,71 @@ package org.bigbluebutton.modules.whiteboard
+         * The following methods handles the presenter typing text into the textbox. The challenge here is how to maintain focus
+         * on the textbox while the presenter changes the size of the font and color.
+         *
+         * The text annotation will have 3 states (CREATED, EDITED, PUBLISHED). When the presenter creates a textbox, the other
+         * users are notified and the text annotation is in the CREATED state. The presenter can then type text, change size, font and
+         * the other users are updated. This is the EDITED state. When the presented hits the ENTER/RETURN key, the text is committed/published.
+         *
+         */
+        public function textObjSpecialListener(event:KeyboardEvent):void {
+            // check for special conditions
+            if (event.keyCode == Keyboard.DELETE || event.keyCode == Keyboard.BACKSPACE || event.keyCode == Keyboard.ENTER) {
+                var sendStatus:String = TextObject.TEXT_UPDATED;
+                var tobj:TextObject = event.target as TextObject;
+                sendTextToServer(sendStatus, tobj);
+            }
+            // stops stops page changing when trying to navigate the text box
+            if (event.keyCode == Keyboard.LEFT || event.keyCode == Keyboard.RIGHT) {
+                event.stopPropagation();
+            }
+        }
+
         public function textObjTextChangeListener(event:Event):void {
             // The text is being edited. Notify others to update the text.
             var sendStatus:String = TextObject.TEXT_UPDATED;
-            var tf:TextObject = event.target as TextObject;  
-            sendTextToServer(sendStatus, tf);  
+            var tf:TextObject = event.target as TextObject;
+            sendTextToServer(sendStatus, tf);
         }
-            
-    public function textObjGainedFocusListener(event:FocusEvent):void {
-//      LogUtil.debug("### GAINED FOCUS ");
+
+        public function textObjGainedFocusListener(event:FocusEvent):void {
+            //      LogUtil.debug("### GAINED FOCUS ");
             // The presenter is ready to type in the text. Maintain focus to this textbox until the presenter hits the ENTER/RETURN key.
             maintainFocusToTextBox(event);
-    }
-    
-    public function textObjLostFocusListener(event:FocusEvent):void {
-//      LogUtil.debug("### LOST FOCUS ");
+        }
+
+        public function textObjLostFocusListener(event:FocusEvent):void {
+            //      LogUtil.debug("### LOST FOCUS ");
             // The presenter is moving the mouse away from the textbox. Perhaps to change the size and color of the text.
             // Maintain focus to this textbox until the presenter hits the ENTER/RETURN key.
             maintainFocusToTextBox(event);
-    }
-    
+        }
+
         private function maintainFocusToTextBox(event:FocusEvent):void {
+			// In some scenarios, quickly creating multiple text zones will make

```

Comment:
It looks like 489-493 are the only actual changes. Is that correct? There's a lot of non-changes. Also, won't this force the text to publish whenever focus is lost? If you have a text open and click the users window I think the text will publish.